File: java/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> hashMap = new HashMap<Integer, Integer>();

        for(int i = 0; i < nums.length; i++) {
            if (hashMap.containsKey(nums[i])) {
                int[] res = { i, hashMap.get(nums[i]) };
                return res;
            } else {
                hashMap.put(target - nums[i], i);
            }
        }
    }
}

File: javascript/1-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
  let hashmap = {};

  for(let i = 0; i < nums.length; i++) {
      if (nums[i] in hashmap) {
          return [i, hashmap[nums[i]]];
      } else {
          hashmap[target - nums[i]] = i;
      }
  }
};
